tests/Oro/Tests/ORM/AST/Query/Functions/fixtures/mysql/cast.yml

```diff
@@ -1,31 +1,42 @@
+#Type::INTEGER
 - functions:
     - { name: "cast", className: "Oro\\ORM\\Query\\AST\\Functions\\Cast", type: "numeric" }
-  dql: "SELECT CAST(12 as int) FROM Oro\\Entities\\Foo f WHERE f.id = 1"
-  sql: "SELECT CAST(12 AS unsigned) AS sclr0 FROM test_foo t0_ WHERE t0_.id = 1"
+  dql: "SELECT CAST('123' as integer) FROM Oro\\Entities\\Foo f WHERE f.id = 1"
+  sql: "SELECT CAST('123' AS signed) AS sclr0 FROM test_foo t0_ WHERE t0_.id = 1"
   expectedResult:
-      - 12
+      - 123
 
+#Type::STRING
 - functions:
     - { name: "cast", className: "Oro\\ORM\\Query\\AST\\Functions\\Cast", type: "numeric" }
-  dql: "SELECT CAST('12' as int) FROM Oro\\Entities\\Foo f WHERE f.id = 1"
-  sql: "SELECT CAST('12' AS unsigned) AS sclr0 FROM test_foo t0_ WHERE t0_.id = 1"
```

Comment:
BC break here. NOTE! This library may be used not only in Oro projects. This is BC break, we should avoid it